Firestore/core/src/firebase/firestore/core/direction.h

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_CORE_DIRECTION_H_
+#define FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_CORE_DIRECTION_H_
+
+#include "absl/base/attributes.h"
+
+namespace firebase {
+namespace firestore {
+namespace core {
+
+/** Interface used for all query orderings. All Directions are immutable. */
+class Direction {
+ public:
+  ABSL_CONST_INIT static const Direction Ascending;
```

Comment:
I haven't been able to make that work. The compiler complains that it can't make incomplete types `constexpr` (and indicates that the type is incomplete until the closing `};`).

If I were willing to put these outside the class this could work, but most of the point of this is to make them scoped so I think this is likely the best form.